Revit_Core_Engine/Modify/Insulation.cs

```diff
@@ -0,0 +1,108 @@
+        [Input("insulationThickness", "Thickness of insulation to be added or updated on the host object.")]
+        [Output("insulation", "Insulation that has been added or updated on the host element.")]
+        public static InsulationLiningBase Insulation(this Element host, ElementType insulationType, double insulationThickness)
+        {
+            if (host == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Insulation could not be created. Host element not found.");
+                return null;
+            }
+            if (insulationType != null && insulationThickness < BH.oM.Geometry.Tolerance.Distance)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Insulation thickness cannot be 0 or negative. To remove existing host insulation, set insulation type as null.");
+                return null;
+            }
+
+            Document doc = host.Document;
+            InsulationLiningBase ins = null;
+            ICollection<ElementId> insIds = InsulationLiningBase.GetInsulationIds(doc, host.Id);
+            insulationThickness = insulationThickness.FromSI(UnitType.UT_Length);
+            BuiltInCategory hostCategory = (BuiltInCategory)host.Category.Id.IntegerValue;
+
+            if (insIds.Count == 0 && insulationType != null)
+            {
+                if (host is Pipe || hostCategory is BuiltInCategory.OST_PipeFitting || hostCategory is BuiltInCategory.OST_PipeAccessory || hostCategory is BuiltInCategory.OST_FlexPipeCurves)
+                {
+                    ins = PipeInsulation.Create(doc, host.Id, insulationType.Id, insulationThickness);
+                }
+                else if (host is Duct || hostCategory is BuiltInCategory.OST_DuctFitting || hostCategory is BuiltInCategory.OST_DuctAccessory || hostCategory is BuiltInCategory.OST_FlexDuctCurves)
+                {
+                    ins = DuctInsulation.Create(doc, host.Id, insulationType.Id, insulationThickness);
+                }
+                else
+                {
+                    BH.Engine.Reflection.Compute.RecordError("Insulation could not be created. Host element type does not support insulation adding functionality.");
+                    return null;
+                }
+            }
+            else if (insIds.Count == 1)
+            {
+                ElementId insId = insIds.ToList()[0];
+                ins = doc.GetElement(insId) as InsulationLiningBase;
+                if (insulationType != null)
+                {
+                    if (ins.GetTypeId() != insulationType.Id)
+                    {
+                        ins.ChangeTypeId(insulationType.Id);
+                    }
+                    
+                    ins.Thickness = insulationThickness;
+                }
+                else
+                {
+                    doc.Delete(insId);
```

Comment:
I think you should return `null` here instead of letting the method progress to  line 103, `return ins`.